pallets/loans/src/functions.rs

```diff
@@ -375,6 +376,11 @@ impl<T: Config> Pallet<T> {
 				// ensure amount is not more than current debt
 				let repay_amount = amount.min(debt);
 
+				let new_repaid_amount = loan_info
+					.repaid_amount
+					.checked_add(&amount)
+					.ok_or(DispatchError::Arithmetic(ArithmeticError::Overflow))?;
+
```

Comment:
> This should be repay_amount instead of amount.
Oops, nicely spotted :)

> But we also simply increment the borrowed amount upon borrow. So for a CreditLine those two values would simply increment.
I think this is fine. We should probably rename these fields, it's really the "total borrowed / repaid amount over time". This will be used by some loan types where it makes sense, while others it will just be ignored and indeed it's not particularly relevant information. I think tracking this only for certain loan types probably adds unnecessary complexity.

Reply:
@offerijns if this bug lived despite the tests, that tells me that the tells are not covering enough :) could you have a look at it?

Reply:
Good point, just added a simple check in the tests that would have caught this.